File: admin-api/src/mujahid_admin/services/metrics.py
from __future__ import annotations

from mujahid_admin.clients.bot import bot_client
from mujahid_admin.schemas.metrics import MetricsSummary
from mujahid_admin.services.metrics_parser import build_metrics_summary, parse_prometheus


async def get_metrics_summary() -> MetricsSummary:
    text = await bot_client.fetch_metrics_text()
    parsed = parse_prometheus(text)
    summary = build_metrics_summary(parsed)
    return MetricsSummary.model_validate(summary)

File: admin-api/src/services/metrics.ts
import { botClient } from "@/clients/bot";
import { buildMetricsSummary, parsePrometheus } from "@/services/metrics-parser";
import type { MetricsSummary } from "@/types";

export async function getMetricsSummary(): Promise<MetricsSummary> {
  const text = await botClient.fetchMetricsText();
  const parsed = parsePrometheus(text);
  return buildMetricsSummary(parsed);
}
